packages/react-menu/src/components/MenuList/useMenuList.ts

```diff
@@ -21,5 +21,29 @@ export const useMenuList = (
     resolveShorthandProps(props, []),
   );
 
+  const { checkedValues, onCheckedValueChange } = state;
+  state.toggleCheckbox = useEventCallback(
+    (e: React.MouseEvent | React.KeyboardEvent, name: string, value: string, checked: boolean) => {
+      const checkedItems = checkedValues?.[name] || [];
+
+      if (checked) {
+        const newCheckedItems = [...checkedItems];
+        newCheckedItems.splice(newCheckedItems.indexOf(value), 1);
+        onCheckedValueChange?.(e, name, newCheckedItems);
+      } else {
+        onCheckedValueChange?.(e, name, [...checkedItems, value]);
+      }
+    },
+    [onCheckedValueChange, checkedValues],
```

Comment:
```suggestion
```

This is confusing, I don't think that we expect any `dependencies` there. I overlooked this in `useEventCallback()` implementation, but we should not have dependencies at all:

```diff
export const useEventCallback = <Args extends unknown[], Return>(
  fn: (...args: Args) => Return,
-  dependencies: React.DependencyList,
) => {
  const callbackRef = React.useRef<typeof fn>(() => {
    throw new Error('Cannot call an event handler while rendering');
  });

  React.useEffect(
    () => {
      callbackRef.current = fn;
    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [fn, ...dependencies],
  );

  return React.useCallback(
    (...args: Args) => {
      const callback = callbackRef.current;
      return callback(...args);
    },
-    [callbackRef],
+    [],
  );
};
```

Reply:
Removed the dep array from `useEventcallback`